File: app/src/main/java/com/thinkzi/weather/ui/di/ViewModelModule.java
package com.thinkzi.weather.ui.di;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import com.thinkzi.weather.ui.viewmodel.WeatherInCityUIModelForFiveDaysViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;

/**
 * Allow us to inject dependencies via constructor injection
 */
@Module
public abstract class ViewModelModule {

    @Binds
    @IntoMap
    @ViewModelKey(WeatherInCityUIModelForFiveDaysViewModel.class)
    abstract ViewModel bindsWeatherInCityUIModelForFiveDaysViewModel(WeatherInCityUIModelForFiveDaysViewModel _weatherInCityUIModelForFiveDaysViewModel);

    @Binds
    abstract ViewModelProvider.Factory bindsViewModelFactory(ViewModelFactory _viewModelFactory);
}

File: app/src/main/java/com/thinkzi/weather/ui/di/ViewModelModule.kt
package com.thinkzi.weather.ui.di

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.thinkzi.weather.ui.viewmodel.WeatherInCityUIModelForFiveDaysViewModel

import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap

/**
 * Allow us to inject dependencies via constructor injection
 */
@Module
abstract class ViewModelModule {

    @Binds
    @IntoMap
    @ViewModelKey(WeatherInCityUIModelForFiveDaysViewModel::class)
    internal abstract fun bindsWeatherInCityUIModelForFiveDaysViewModel(
        _weatherInCityUIModelForFiveDaysViewModel: WeatherInCityUIModelForFiveDaysViewModel
    ): ViewModel

    @Binds
    internal abstract fun bindsViewModelFactory(_viewModelFactory: ViewModelFactory): ViewModelProvider.Factory
}
